File: C#/第一部分 C#语言/第三章 程序的控制结构/第三节 switch语句/switch.cs
// 由bhw设计
/*
第3章 第3节 switch语句

一、适用场景
1) 离散值分支判断（菜单、枚举、状态码）。
2) 可读性通常优于长 if-else 链。

二、原理
1) switch 按值匹配 case。
2) C# 传统 switch case 需要 break 防止继续执行。

三、习题
1) 输入 1~7 输出星期。
2) 输入字符 '+' '-' '*' '/' 实现简易计算器。

四、答案与讲解
1) 每个数字对应一个 case。
2) 计算器里要特别处理除数为 0。
*/
using System; // 控制台输入输出

class Program // 类定义
{
    static void Main() // 主函数
    {
        Console.Write("请输入 1~7: "); // 提示输入
        string text = Console.ReadLine() ?? ""; // 读取输入
        if (!int.TryParse(text, out int day)) // 解析整数
        {
            Console.WriteLine("输入错误"); // 错误提示
            return; // 结束
        }

        switch (day) // 按 day 分支
        {
            case 1: Console.WriteLine("周一"); break; // case 1
            case 2: Console.WriteLine("周二"); break; // case 2
            case 3: Console.WriteLine("周三"); break; // case 3
            case 4: Console.WriteLine("周四"); break; // case 4
            case 5: Console.WriteLine("周五"); break; // case 5
            case 6: Console.WriteLine("周六"); break; // case 6
            case 7: Console.WriteLine("周日"); break; // case 7
            default: Console.WriteLine("输入非法"); break; // 默认分支
        }
    } // Main 结束
} // 类结束
// 由bhw设计

File: C++/第一部分 C++语法/第三章 程序控制结构/第二节 switch语句/switch.cpp
// 由bhw设计
/*
第三章 第二节 switch语句

一、适用场景
1) 一个变量有多个“离散值”分支时，switch通常比多重if更清晰。
2) case后面一般是常量表达式。

二、关键字含义
1) switch(表达式)：根据表达式的值匹配case。
2) case 常量: 对应分支。
3) break: 跳出switch，防止继续向下贯穿。
4) default: 兜底分支，前面都不匹配时执行。

三、练习题
1) 输入1~7输出星期几。
2) 输入运算符(+ - * /)并计算两个整数。
*/
#include <iostream> // 输入输出
using namespace std; // 标准命名空间

int main() // 主函数
{
    int day = 0; // 存储星期编号
    cout << "请输入1~7: "; // 提示输入
    cin >> day; // 读取day

    switch (day) // 开始多分支匹配
    {
    case 1: cout << "星期一" << endl; break; // 匹配1
    case 2: cout << "星期二" << endl; break; // 匹配2
    case 3: cout << "星期三" << endl; break; // 匹配3
    case 4: cout << "星期四" << endl; break; // 匹配4
    case 5: cout << "星期五" << endl; break; // 匹配5
    case 6: cout << "星期六" << endl; break; // 匹配6
    case 7: cout << "星期日" << endl; break; // 匹配7
    default: cout << "输入无效" << endl; break; // 兜底
    }

    return 0; // 程序结束
} // main结束

/*
练习题答案（注释版）
1) 星期映射可直接复用上面switch结构。
2) 运算器示例（注释版）：

// int a, b; // 两个整数
// char op; // 运算符
// cin >> a >> op >> b; // 输入格式如 6 + 2
// switch (op) // 根据运算符分支
// {
// case '+': cout << a + b << endl; break; // 加
// case '-': cout << a - b << endl; break; // 减
// case '*': cout << a * b << endl; break; // 乘
// case '/': if (b != 0) cout << a / b << endl; else cout << "除数不能为0" << endl; break; // 除
// default: cout << "未知运算符" << endl; break; // 兜底
// } // 结束
*/
// 由bhw设计
